docs/automated-testing/e2e-testing/recipes/postman-testing.md

```diff
@@ -0,0 +1,230 @@
+
+    ```javascript
+    // Please treat this as pseudocode, and adjust as necessary.
+
+    /* The request to an oauth2 authorization endpoint that will issue a token 
+    based on provided credentials.*/
+    const oauth2Request = POST {...};
+    var getToken = true;
+    if (pm.environment.get('ACCESS_TOKEN_EXPIRY') <= (new Date()).getTime()) {
+        console.log('Token is expired')
+    } else {
+        getToken = false;
+        console.log('Token and expiry date are all good');
+    }
+    if (getToken === true) {
+        pm.sendRequest(oauth2Request, function (_, res) {
+                console.log('Save the token')
+                var responseJson = res.json();
+                pm.environment.set('token', responseJson.access_token)
+                console.log('Save the expiry date')
+                var expiryDate = new Date();
+                expiryDate.setSeconds(expiryDate.getSeconds() + responseJson.expires_in);
+                pm.environment.set('ACCESS_TOKEN_EXPIRY', expiryDate.getTime());
+        });
+    }
+    ```
+
+6. Use Postman IDE to exercise endpoints.
+7. Export collection and environment files then remove any secrets before committing to your repo.
+
+Starting with this approach has the following upsides:
+
+- You've set yourself up for the beginning stages of an E2E postman collection by aggregating the collections into a single file and using environment files to make it easier to switch environments.
+- Token is refreshed automatically on every call in the collection. This saves you time normally lost from manually having to request a token that expired.
+- Grants QA/Dev granular control of submitting combinations of input data per endpoint.
+- Grants devs a common experience via Postman IDE features.
+
+Ending with this approach has the following downsides:
+
+- Promotes unsafe sharing of secrets. Credentials needed to request JWT token in the prefetch script are being manually shared.
+- Secrets may happen to get exposed in the git commit history for various reasons (ex. Sharing the exported Postman Environment files).
+- Collections can only be used locally to hit APIs (local or deployed). Not CI based.
+- Each developer has to keep both their Postman Collection and Postman environment file(s) updated in order to keep up with latest changes to deployed services.
+
+### Use Case - Hands-on Functional Testing Of Endpoints with Azure Key Vault and Azure App Config
+
+A developer or qa analyst may have an existing API test suite of Postman Collections, however, they now want to discourage unsafe sharing of secrets. As a result, they build a script that connects to both Keyvault and Azure App Config in order to automatically generate Postman Environment files instead of checking them into a shared repository.
```

Comment:
```suggestion
A developer or QA analyst may have an existing API test suite of Postman Collections, however, they now want to discourage unsafe sharing of secrets. As a result, they build a script that connects to both Key Vault and Azure App Config in order to automatically generate Postman Environment files instead of checking them into a shared repository.
```